services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/BasicScanTest.java

```diff
@@ -140,6 +141,24 @@ public void scanAllRecordsDefaultSettings() {
         assertThat(page.lastEvaluatedKey(), is(nullValue()));
     }
 
+    @Test
+    public void queryAllRecordsDefaultSettings_withProjection() {
```

Comment:
Can we add a test for multiple projected attributes? Also given what we know about the implementation I think I would sleep better if we also had a test with projected attributes AND a filter expression together, perhaps in a configuration that is likely to require them to merge correctly.